bitcoin/src/address.rs

```diff
@@ -693,6 +759,32 @@ impl Address {
         format!("{}:{:#}", schema, self)
     }
 
+    /// Returns true if the given pubkey is directly related to the address payload.
+    ///
+    /// This is determined by directly comparing the address payload with either the
+    /// hash of the given public key or the segwit redeem hash generated from the
+    /// given key. For taproot addresses, the supplied key is assumed to be tweaked
+    pub fn is_related_to_pubkey(&self, pubkey: &PublicKey) -> bool {
+        let pubkey_hash = pubkey.pubkey_hash();
+        let payload = self.payload.as_bytes();
+        let xonly_pubkey = XOnlyPublicKey::from(pubkey.inner);
+
+        (*pubkey_hash == *payload)
+            || (xonly_pubkey.serialize() == *payload)
+            || (*segwit_redeem_hash(&pubkey_hash) == *payload)
+    }
+
+    /// Returns true if the supplied xonly public key can be used to derive the address.
+    ///
+    /// This will only work for Taproot addresses. The Public Key is
+    /// assumed to have already been tweaked.
+    pub fn is_related_to_xonly_pubkey(&self, xonly_pubkey: &XOnlyPublicKey) -> bool {
```

Comment:
I wonder if these two are safe to use with `UncheckedNetwork`. Will need to think about it. Any argument?

Reply:
Heh, but for *these* I think it does make sense to use them with `UncheckedNetwork`, because they're questions about ownership/signability that aren't necessarily related to the networck that the address should be used on.

Reply:
I have not copied any other methods from `NetworkChecked` to `NetworkUnchecked`. I believe we can leave them on `NetworkChecked` now, copying to `NetworkUnchecked`, if/when needed, should not introduce problems.